x-pack/plugins/apm/public/application/action_menu/index.tsx

```diff
@@ -0,0 +1,126 @@
+  const { search } = window.location;
+  const { application, http } = core;
+  const { basePath } = http;
+  const { capabilities } = application;
+  const canAccessML = !!capabilities.ml?.canAccessML;
+  const {
+    isAlertingAvailable,
+    canReadAlerts,
+    canSaveAlerts,
+    canReadAnomalies,
+  } = getAlertingCapabilities(plugins, capabilities);
+
+  function apmHref(path: string) {
+    return getAPMHref({ basePath, path, search });
+  }
+
+  function kibanaHref(path: string) {
+    return basePath.prepend(path);
+  }
+
+  return (
+    <EuiHeaderLinks>
+      <EuiHeaderLink
+        color="primary"
+        href={apmHref('/settings')}
+        iconType="gear"
+      >
+        {i18n.translate('xpack.apm.settingsLinkLabel', {
+          defaultMessage: 'Settings',
+        })}
+      </EuiHeaderLink>
+      {isAlertingAvailable && (
+        <AlertingPopoverAndFlyout
+          basePath={basePath}
+          canReadAlerts={canReadAlerts}
+          canSaveAlerts={canSaveAlerts}
+          canReadAnomalies={canReadAnomalies}
+          includeTransactionDuration={serviceName !== undefined}
+        />
+      )}
+      {canAccessML && <AnomalyDetectionSetupLink />}
+      <EuiHeaderLink
+        color="primary"
+        href={kibanaHref('/app/home#/tutorial/apm')}
+        iconType="indexOpen"
+      >
+        {i18n.translate('xpack.apm.addDataButtonLabel', {
+          defaultMessage: 'Add data',
+        })}
+      </EuiHeaderLink>
+    </EuiHeaderLinks>
+  );
+}
+
+/**
+ * Get the mount point to set the header menu with the header links.
+ *
+ * Alerts and ML links require a lot of context so include what we need here.
+ */
+export function getActionMenuMountPoint(
+  apmPluginContextValue: ApmPluginContextValue,
+  serviceName?: string
+) {
+  const { appMountParameters, core, plugins } = apmPluginContextValue;
+
+  return toMountPoint(
+    <Router history={appMountParameters.history}>
+      <RedirectAppLinks application={core.application}>
+        <ApmPluginContext.Provider value={apmPluginContextValue}>
+          <UrlParamsProvider>
+            <LicenseProvider>
+              <ServiceNameContext.Provider value={serviceName}>
+                <AlertsContextProvider
+                  value={{
+                    actionTypeRegistry:
+                      plugins.triggersActionsUi.actionTypeRegistry,
+                    alertTypeRegistry:
+                      plugins.triggersActionsUi.alertTypeRegistry,
+                    capabilities: core.application.capabilities,
+                    docLinks: core.docLinks,
+                    http: core.http,
+                    toastNotifications: core.notifications.toasts,
+                  }}
+                >
+                  <ActionMenu />
+                </AlertsContextProvider>
+              </ServiceNameContext.Provider>
+            </LicenseProvider>
+          </UrlParamsProvider>
+        </ApmPluginContext.Provider>
+      </RedirectAppLinks>
+    </Router>
```

Comment:
instead of replicating the whole APM app, why not use https://github.com/httptoolkit/react-reverse-portal to solve this issue?

Reply:
Never heard of that. I could try it.

Reply:
Yeah it's pretty cool, basically you create a portal node in render app function, pass that to render action function, and create and InPortal there, and also pass that portal node to ApmApp inside context etc create and OutPortal with same portal node and that's it